Core/src/main/java/com/plotsquared/core/events/PlayerPlotChatEvent.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * PlotSquared, a land and world management plugin for Minecraft.
+ * Copyright (C) IntellectualSites <https://intellectualsites.com>
+ * Copyright (C) IntellectualSites team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.plotsquared.core.events;
+
+import com.plotsquared.core.player.PlotPlayer;
+import com.plotsquared.core.plot.Plot;
+
+import java.util.Set;
+
+/**
+ * @since TODO
+ */
+public class PlayerPlotChatEvent extends PlotEvent implements CancellablePlotEvent {
+
+    private final PlotPlayer<?> player;
+    private final String message;
```

Comment:
Maybe rather use an adventure component for the message?
The passed message as a string is before the translation of color codes etc happened. Would be more suitable to use the component which is used as the "msg" placeholder.